market/ipni/ipni-provider/ipni-provider.go

```diff
@@ -73,6 +74,7 @@ type Provider struct {
 	indexStore    *indexstore.IndexStore
 	sc            *chunker.ServeChunker
 	keys          map[string]*peerInfo // map[peerID String]Private_Key
+	latest        map[string]cid.Cid   // map[peerID String]last published head, used to avoid duplicate announce
```

Comment:
the changes in here wrt `latest` look like bugfixes, or is it just the fact that 30 seconds now make this task more onerous so it's worth being spare with what we provide?